supplemental_notebooks/2021-09-20_muddiest_points.Rmd

```diff
@@ -0,0 +1,176 @@
+
+### CRAN packages
+
+Most general use packages (not biology-specific) are available through CRAN, the Comprehensive R Archive Network.
+To install a package from CRAN, you use the `install.packages()` function, which will install the requested package(s) as well as any packages they might depend on.
+So if I wanted to install (or update) the `tidyverse` package and all of its dependencies, I would use the following command:
+
+```{r, eval = FALSE}
+install.packages("tidyverse")
+```
+
+If I had more than one package to install, I might use a vector of the package names (each enclosed by quotes):
+```{r eval = FALSE}
+install.packages(c("magrittr", "dplyr", "palmerpenguins"))
+```
+
+The first time you use `install.packages()` you may be prompted (in the Console) to choose a "mirror". 
+You will need to enter any responses in the console.
+Using option `0`, the "cloud", is usually safe, but you can choose another location if you prefer.
+
+If you are using RStudio, there is also an "Install" button in the `Packages` tab, which will prompt you to enter the packages you want to install, but this will only work with CRAN packages.
+
+### Bioconductor packages
+
+To install [Bioconductor](https://bioconductor.org/) packages, you need to use a specialized installation function from the `BiocManager` package.
+So the first step is to install that package from CRAN:
+
+```{r eval = FALSE}
+install.packages("BiocManager")
+```
+
+Once that is installed, you can use `BiocManager::install()` to install any packages that are part of Bioconductor and their dependencies. 
+(In fact you could use this always: it will find CRAN packages as well!)
+
+```{r, eval = FALSE}
+BiocManager::install("DESeq2")
+```
+
+`BiocManager::install()` tends to be a bit more chatty than the base R installation; be on the lookout for prompts that you need to respond to in the console.
+
+## Logical vectors
+
+I briefly talked about logical (Boolean) vectors during the first session, but they can definitely be confusing at times, so here I will go into a bit more depth. 
+
+First off, what is a logical vector? A logical vector is one that contains only the values `TRUE` and `FALSE` (and potentially `NA`).
+Notice the all caps: R requires this. 
+
+So the simplest way to create a logical vector is just like another vector, using `c()` but only those values:
+
+```{r}
+my_bool <- c(TRUE, FALSE, FALSE, TRUE)
+my_bool
+class(my_bool)
+```
+(Full disclosure, you _can_ use `T` and `F` in addition to `TRUE` and `FALSE`, but this is a bad idea, because it is possible to overwrite those abbreviations like normal variable names. `F <- TRUE` is a perfectly valid statement, even though it can wreak havoc on sloppy code.)
+
+
+That isn't too useful, generally, because we usually want to create our logical vectors based on some data. To that we will tend to use one of a few comparison or logical operators:
+
+Starting wiht the comparisons:
+* `==`: equal to
```

Comment:
I think you need a newline for this render properly:

```suggestion
Starting wiht the comparisons:

* `==`: equal to
```

Reply:
Oh also a typo
```suggestion
Starting with the comparisons:

* `==`: equal to
```

Reply:
fixed!